rusty-machine/src/linalg/matrix/mod.rs

```diff
@@ -1128,8 +1130,12 @@ impl<'a, T: Copy + One + Zero + Add<T, Output = T>> Add<T> for &'a Matrix<T> {
 impl<'a, T: Copy + One + Zero + Add<T, Output = T>> Add<&'a T> for Matrix<T> {
     type Output = Matrix<T>;
 
-    fn add(self, f: &T) -> Matrix<T> {
-        (&self) + f
+    fn add(mut self, f: &T) -> Matrix<T> {
+        for val in self.data.iter_mut() {
+            *val = *val + *f;
+        }
+
+        self
```

Comment:
You could use your `apply` function here (and in other `Matrix` op scalar functions)